clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -128,13 +127,6 @@ private void changeSubscription(Set<String> topicsToSubscribe) {
         if (!this.subscription.equals(topicsToSubscribe)) {
             this.subscription = topicsToSubscribe;
             this.groupSubscription.addAll(topicsToSubscribe);
-
-            // Remove any assigned partitions which are no longer subscribed to
-            for (Iterator<TopicPartition> it = assignment.keySet().iterator(); it.hasNext(); ) {
```

Comment:
I believe this method is not subsequently called in `assignFromSubscribed` or `assignFromUser`. Plus, it seems there are already existing unit tests in `SubscriptionStateTest` that verify that the assignment is updated only when necessary. Here are examples for [`assignFromUser`](https://github.com/vahidhashemian/kafka/blob/f60fb52a702a6bcb854e9560548174fc5ad86730/clients/src/test/java/org/apache/kafka/clients/consumer/internals/SubscriptionStateTest.java#L49), [`assignFromSubscribed`](https://github.com/vahidhashemian/kafka/blob/f60fb52a702a6bcb854e9560548174fc5ad86730/clients/src/test/java/org/apache/kafka/clients/consumer/internals/SubscriptionStateTest.java#L92), and [`subscribe`](https://github.com/vahidhashemian/kafka/blob/f60fb52a702a6bcb854e9560548174fc5ad86730/clients/src/test/java/org/apache/kafka/clients/consumer/internals/SubscriptionStateTest.java#L82). There is one missing for `subscribeFromPattern`, that I could add after [this line](https://github.com/vahidhashemian/kafka/blob/f60fb52a702a6bcb854e9560548174fc5ad86730/clients/src/test/java/org/apache/kafka/clients/consumer/internals/SubscriptionStateTest.java#L159). Let me know if I'm missing further unit tests that are required. Thanks.